### PR TITLE
Edit Rota

### PR DESCRIPTION
**Pull request by narayanvaishali**
Please review the code.

Known issue

- Data loading while view and edit is extremely slow
- Data needs restructuring. Currently there are 2 records for each date per shift. So its not displaying correctly. Data saved correctly

- look and feel of drop downs in edit mode needs to be fixed

### CODE REVIEW

**Comment by ayrbox** [comment]
## Rota Loading...
It seems to be working but the rending take very longs time. The rota (both view and edit) must require  optimisation while rendering.

File: src/db/auth.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    .createUserWithEmailAndPassword(email, pw)
+    .then(saveUser);
+}
+
+export function logout() {
+  return firebaseAuth()
+    .signOut()
+    .then(res => {
+      localStorage.removeItem("token");
+    }); bra
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/db/auth.js]: Lol ..Sorry typo !!

File: src/views/MonthlyView/MonthlyView.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+import EditMonthlyView from './EditMonthlyView';
+import { database } from '../../db';
+
+
+var clonedArr= [];
+const Child = () => (
+<div className='modal'>
+      Hello, World!
+  </div>
+)
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/views/MonthlyView/MonthlyView.js]: Removed

File: src/views/MonthlyView/MonthlyView.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+const LayoutDisplay = () => (
+<div className='modal'>
+      Hello, World!
+  </div>
+)
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/views/MonthlyView/MonthlyView.js]: Removed

File: src/views/Rota/Rota.1.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/views/Rota/Rota.1.js]: Removed

File: package.json [hunk 5]
```diff
@@ -11,8 +11,11 @@
     "prop-types": "^15.6.2",
     "react": "^16.6.3",
     "react-dom": "^16.6.3",
+    "react-redux": "^6.0.0",
     "react-router-dom": "^4.3.1",
-    "react-scripts": "2.1.1"
+    "react-scripts": "2.1.1",
+    "react-select": "^2.1.2",
+    "redux": "^4.0.1"
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply package.json]: Removed

**Comment by narayanvaishali** [comment]
This is known issue (as listed in my comments)

File: src/containers/Shift/index.js [hunk 6]
```diff
@@ -18,28 +18,44 @@ class ShiftCell extends Component {
...
+
+          //console.log('sch ' + Object.keys(sch));
+          const list = Object.keys(sch).map(id => {
+          //  console.log('sch ' + sch[id].date);
+          let _key = id;
+          let _date  = sch[id].date;
+          let _day  = sch[id].day;
+          let  _name =sch[id].name;
+          let  _shift_AM = sch[id].shift_AM; 
+          let _shift_PM  = sch[id].shift_PM;
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/containers/Shift/index.js]: do you mean not recommended?

File: src/db/auth.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    .then(() => user);
+}
+
+export function isAuthenticated() {
+  if (localStorage.token) {
+    //@TODO: Use token similar to JWT so that user information can be displayed
+
+    return localStorage.token;
+  }
+}
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/db/auth.js]: removed

**Comment by ayrbox** [comment]
@narayanvaishali  ??

File: src/views/MonthlyView/MonthlyView.js [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+class MonthlyView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      month: new Date().getMonth(),
+      year: new Date().getFullYear(),
+      isEdit : false,
+      show : 'V',
+      newscheduledata : []
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/views/MonthlyView/MonthlyView.js]: will do this later

File: src/components/Register.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  invertedBtn: {
+    color: "#fff",
+    backgroundColor: "transparent",
+    border: "2px #fff solid",
+    boxShadow: "none"
+  },
+  white: {
+    color: "#fff"
+  }
+});
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/components/Register.js]: moved to style.js

File: src/containers/Shift/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/containers/Shift/index.js]: removed

File: src/views/MonthlyView/EditMonthlyView.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+              <RotaGrid users={staffList} days={days} show ='E'/>
+            </div>
+          </Grid>
+        </Paper>
+      </Layout>
+    );
+  }
+}
+
+export default withStyles(styles)(EditMonthlyView);
```
[the remark this answers is not in the document]
Comment (narayanvaishali) [reply src/views/MonthlyView/EditMonthlyView.js]: may be later, will amend later

**Pull request merged**